src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -112,16 +115,45 @@ public class SftpClient {
    * @param host the host address
    * @param port the remote connection port
    */
-  public SftpClient(String host, int port, PRNGAlgorithm prngAlgorithm, SchedulerService schedulerService) {
+  public SftpClient(String host, int port, PRNGAlgorithm prngAlgorithm, SchedulerService schedulerService, boolean kexHeader) {
```

Comment:
We are in the internal package, so there is no problem with changing it, but usually, when it comes to changing a public method, I prefer to keep that signature and make a new constructor adding the new value
Example:
Before:
```
public SftpClient(String x){
  this.x=x;
}
```
After:
```
public SftpClient(String x){
  SftpClient(x, null)
}
public SftpClient(String x, Integer y){
  this.x=x;
  this.y=y;
}
```

Reply:
like  public SftpClient(String host, int port, PRNGAlgorithm prngAlgorithm, SchedulerService schedulerService) {
   this(host,port,prngAlgorithm,schedulerService,true);
  } or  public SftpClient(String host, int port, PRNGAlgorithm prngAlgorithm, SchedulerService schedulerService) {
    new SftpClient(host, port, prngAlgorithm,schedulerService,true);} because i have to put new because it's expecting the method call. I used true as default for strict kex